vmdk_plugin/drivers/photon/photon_driver.go

```diff
@@ -113,25 +113,25 @@ func NewVolumeDriver(targetURL string, projectID string, hostID string, mountDir
 // map hasn't been initialized, return 1 prevent detach and remove.
 
 // Return the number of references for the given volume
-func (d *VolumeDriver) getRefCount(vol string) uint {
+func (d *VolumeDriver) getRefCount(vol string) (uint, error) {
 	if d.refCounts.GetInitSuccess() != true {
-		return 1
+		return 0, fmt.Errorf(plugin_utils.ErrorPluginInit)
 	}
-	return d.refCounts.GetCount(vol)
+	return d.refCounts.GetCount(vol), nil
 }
 
 // Increment the reference count for the given volume
-func (d *VolumeDriver) incrRefCount(vol string) uint {
+func (d *VolumeDriver) incrRefCount(vol string) (uint, error) {
 	if d.refCounts.GetInitSuccess() != true {
-		return 1
+		return 0, fmt.Errorf(plugin_utils.ErrorPluginInit)
 	}
-	return d.refCounts.Incr(vol)
+	return d.refCounts.Incr(vol), nil
 }
 
 // Decrement the reference count for the given volume
 func (d *VolumeDriver) decrRefCount(vol string) (uint, error) {
 	if d.refCounts.GetInitSuccess() != true {
-		return 1, nil
+		return 0, fmt.Errorf(plugin_utils.ErrorPluginInit)
 	}
 	return d.refCounts.Decr(vol)
```

Comment:
-1 is leading to a lot of change of refcount apis. besides, actual refcount of a volume is stored as uint. using -1 is causing underflow error. So using error instead. Also it makes the logic more clear.